File: coupon-service/src/main/java/com/poorbet/couponservice/domain/BetType.java
package com.poorbet.couponservice.domain;

import com.poorbet.commons.rabbit.events.match.dto.MatchResultEventDto;

public enum BetType {
    HOME_WIN,
    DRAW,
    AWAY_WIN,
    OVER_2_5,
    UNDER_2_5,
    OVER_3_5,
    UNDER_3_5;


    public BetStatus mapToStatus(MatchResultEventDto result, int homeGoals, int awayGoals) {
        if (result == null) return BetStatus.PENDING;

        int totalGoals = homeGoals + awayGoals;

        return switch (this) {
            case HOME_WIN -> homeGoals > awayGoals ? BetStatus.WON : BetStatus.LOST;
            case AWAY_WIN -> awayGoals > homeGoals ? BetStatus.WON : BetStatus.LOST;
            case DRAW -> homeGoals == awayGoals ? BetStatus.WON : BetStatus.LOST;
            case OVER_2_5 -> totalGoals > 2 ? BetStatus.WON : BetStatus.LOST;
            case UNDER_2_5 -> totalGoals <= 2 ? BetStatus.WON : BetStatus.LOST;
            case OVER_3_5 -> totalGoals > 3 ? BetStatus.WON : BetStatus.LOST;
            case UNDER_3_5 -> totalGoals <= 3 ? BetStatus.WON : BetStatus.LOST;
        };
    }
}

File: frontend/src/app/shared/types/bet-type.ts
export enum BetType {
  HomeWin = 'HOME_WIN',
  Draw = 'DRAW',
  AwayWin = 'AWAY_WIN',
  Over2_5 = 'OVER_2_5',
  Under2_5 = 'UNDER_2_5',
  Over3_5 = 'OVER_3_5',
  Under3_5 = 'UNDER_3_5',
}

const GOAL_MARKET_BET_TYPES: ReadonlySet<BetType> = new Set([
  BetType.Over2_5,
  BetType.Under2_5,
  BetType.Over3_5,
  BetType.Under3_5,
]);

export function isGoalMarketBetType(betType: BetType): boolean {
  return GOAL_MARKET_BET_TYPES.has(betType);
}
